mender.go

```diff
@@ -318,6 +323,27 @@ func (m *mender) UploadLog(update UpdateResponse, logs []LogEntry) menderError {
 	return nil
 }
 
+func (m *mender) StartLogWriting(deploymentID string) error {
+	// check if there is id from last deployment (we coudl do reboot)
+	if deploymentID == "" {
```

Comment:
So all of this could actually be done in `UpdateFetchState` and passed along as some sort of update context between the update states. Disable logs right after leaving reboot, reopen when entering update commit state.